### PR TITLE
gen-job.sh fixed based on indentation parameter - and generated files

### PR DESCRIPTION
**Pull request by asim-reza**
[error unmarshaling config/jobs/kubernetes-csi/csi-driver-host-path/csi-driver-host-path-config.yaml: error unmarshaling JSON: while decoding JSON: json: unknown field "resources" #27438](https://github.com/kubernetes/test-infra/issues/27438)

as discussed in the aforementioned issue and[ PR](https://github.com/kubernetes/test-infra/pull/27477), the script was generating invalid indentations for resource assignments the issue has been fixed in the script and the fixed files are added to this as well...

### CODE REVIEW

**Comment by asim-reza** [comment]
@pohly

File: config/jobs/kubernetes-csi/gen-jobs.sh [hunk 1]
```diff
@@ -124,27 +124,26 @@ find "$base" -name '*.yaml' -exec grep -q 'generated by gen-jobs.sh' '{}' \; -de
 # Resource usage of a job depends on whether it needs to build Kubernetes or not.
 resources_for_kubernetes () {
     local kubernetes="$1"
+    local indentation="$2"
 
     # "master" is always built from source. The experimental version might be built from
     # source, depending on what is in the prow.sh script. Here we play it safe and
     # request more resource when testing the experimental version, just in case.
+    echo "${indentation}resources:"
+    echo "${indentation} requests:"
```
Comment (pohly) [inline config/jobs/kubernetes-csi/gen-jobs.sh:133]: I think it's clearer to duplicate these two lines below, as it was before. For example, then it is obvious that `memory` and `cpu` are indented properly.

**Review by pohly — CHANGES_REQUESTED** [review]
Please try to avoid reopening PRs just because you made changes to your branch. Just edit your branch, use `git rebase -i`, then force push with `git push <your repo> +<branch>`.

Unless a merge conflict forces you to rebase onto master, then always rebase onto the same base revision. Then GitHub will show only your changes between force-pushes.

Alternatively, you can also keep adding fixup commits and never force-push until your reviewer tells you to clean up. I personally prefer PRs which get cleaned up right away, before review.

File: config/jobs/kubernetes-csi/gen-jobs.sh (hunk 1 above)
Comment (asim-reza) [reply config/jobs/kubernetes-csi/gen-jobs.sh]: done!

File: config/jobs/kubernetes-csi/gen-jobs.sh [hunk 2]
```diff
@@ -124,27 +124,28 @@ find "$base" -name '*.yaml' -exec grep -q 'generated by gen-jobs.sh' '{}' \; -de
...
     # request more resource when testing the experimental version, just in case.
     case $kubernetes in master|$experimental_k8s_version|release-*) cat <<EOF
-      resources:
-        requests:
-          # these are both a bit below peak usage during build
-          # this is mostly for building kubernetes
-          memory: "9000Mi"
-          # during the tests more like 3-20m is used
-          cpu: 2000m
+    resources:
```
Comment (pohly) [inline config/jobs/kubernetes-csi/gen-jobs.sh:133]: Now that `resources` is in the body of the text, it is visible that it is already indented. That makes `$indention` more like `$additional_indention`.

I think it is cleaner to let `resources` start in the first column (i.e. no indention) and then choose the amount of spaces entirely where `resources_for_kubernetes` is called.

File: config/jobs/kubernetes-csi/gen-jobs.sh [hunk 3]
```diff
@@ -410,7 +411,7 @@ EOF
         # docker-in-docker needs privileged mode
         securityContext:
           privileged: true
-$(resources_for_kubernetes "$kubernetes")
+$(resources_for_kubernetes "$kubernetes" "    ")
```
Comment (pohly) [inline config/jobs/kubernetes-csi/gen-jobs.sh:414]: To expand on the previous point: why is this the right amount of spaces? This becomes more obvious when it matches the number of spaces before `securityContext`.

**Review by pohly — CHANGES_REQUESTED** [review]
Please squash all changes into a single commit before the next update.

File: config/jobs/kubernetes-csi/gen-jobs.sh (hunk 2 above)
Comment (asim-reza) [reply config/jobs/kubernetes-csi/gen-jobs.sh:133]: that makes complete sense ... on it!

Comment (asim-reza) [reply config/jobs/kubernetes-csi/gen-jobs.sh]: Done!

File: config/jobs/kubernetes-csi/gen-jobs.sh (hunk 3 above)
Comment (asim-reza) [reply config/jobs/kubernetes-csi/gen-jobs.sh]: 👍🏼 done

File: config/jobs/kubernetes-csi/gen-jobs.sh [hunk 4]
```diff
@@ -124,27 +124,28 @@ find "$base" -name '*.yaml' -exec grep -q 'generated by gen-jobs.sh' '{}' \; -de
 # Resource usage of a job depends on whether it needs to build Kubernetes or not.
 resources_for_kubernetes () {
     local kubernetes="$1"
-
+    local indentation="$2"
+    local IFS=""
```
[the remark this answers is not in the document]
Comment (asim-reza) [reply config/jobs/kubernetes-csi/gen-jobs.sh:128]: Yes or the `read` ignores leading spaces... That forces `resources:`, `memory:`, `cpu:` all in one indentation level

[the remark this answers is not in the document]
Comment (pohly) [reply config/jobs/kubernetes-csi/gen-jobs.sh:128]: Good point. Let's switch to the approach with `sed` then?

[the remark this answers is not in the document]
Comment (asim-reza) [reply config/jobs/kubernetes-csi/gen-jobs.sh]: done! works perfectly... I dont understand how this works tho 😅

[the remark this answers is not in the document]
Comment (asim-reza) [reply config/jobs/kubernetes-csi/gen-jobs.sh]: does `^` indicate the start of the line?

[the remark this answers is not in the document]
Comment (pohly) [reply config/jobs/kubernetes-csi/gen-jobs.sh]: Yes.

**Pull request merged**